src/training.py

```diff
@@ -93,15 +157,30 @@ def training_step(self, batch, batch_idx):
 
     def validation_step(self, batch, batch_idx):
         loss, embeddings = self._forward_batch(batch, return_embeddings=True)
-        # save predictions for visualization
+        emb_np = embeddings.cpu().detach().numpy()
+        evt_ids_np = batch[1].cpu().numpy()
+        cluster_preds = self.clustering.cluster_and_link(emb_np, evt_ids_np)
+
+        metrics_results = {}
+
+        for metric in self.metrics:
+            if isinstance(metric, (F1ScoreMetric, PrecisionScoreMetric, RecallScoreMetric, AccuracyScoreMetric)):
```

Comment:
reformat long string please to make it shorter, you can spread it over multiple lines